corehq/apps/dump_reload/sql/filters.py

```diff
@@ -86,11 +89,29 @@ def __init__(self, user_id_field, include_web_users=True):
         super().__init__(user_id_field, None)
         self.include_web_users = include_web_users
 
-    def get_ids(self, domain_name):
+    def get_ids(self, domain_name, db_alias=None):
         from corehq.apps.users.dbaccessors import get_all_user_ids_by_domain
         return get_all_user_ids_by_domain(domain_name, include_web_users=self.include_web_users)
 
 
+class CaseIDFilter(IDFilter):
+    def __init__(self, case_field='case'):
+        super().__init__(case_field, None)
+
+    def count(self, domain_name):
+        active_case_count = len(CommCareCase.objects.get_case_ids_in_domain(domain_name))
+        deleted_case_count = len(CommCareCase.objects.get_deleted_case_ids_in_domain(domain_name))
```

Comment:
This loads all of the domain's case ids into memory, which I think we want to avoid. It would be better to use a `.count()` query per shard. This could be implemented as `CommCareCaseManager.count_cases_in_domain(domain_name, include_deleted=True)`.

The new manager method could simply raise `NotImplementedError` if `include_deleted` is false since there is no use case for that branch at this time.

Reply:
If this is just for progress, there is also the `corehq.sql_db.util.estimate_row_count` function which uses the query plan to get an estimated count.

Reply:
Yeah this was lazy on my part. Looking closer at where this used, the Builder object references it [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/dump_reload/sql/filters.py#L132-L136), but I don't see where the builder object's count method is called, and if I set a breakpoint in that method and run `dump_domain_data` it doesn't get hit. So I'm tempted to just remove this `count` method altogether, but can dig a bit more to see if we made an intentional change to the StatsCounter at some point instead.

Reply:
So the count method was introduced after the StatsCounter object, and it looks like in the context of ICDS https://github.com/dimagi/commcare-hq/pull/28895. Is it possible this `count` code isn't currently applicable to `dump_domain_data` because there was only custom ICDS code that took advantage of it?

Reply:
ahhh it is used in `print_domain_stats`

Reply:
So the `count` should return the count of objects the CaseIDFilter is setup for, not the count of cases. This made it a bit trickier, but I updated the filter in 6696b9b3369eb45094b0493f41c67454466710d7 to handle this and added tests to verify that behavior.